src/en/readcomic/src/eu/kanade/tachiyomi/extension/en/readcomic/Readcomic.kt

```diff
@@ -0,0 +1,199 @@
+package eu.kanade.tachiyomi.extension.en.readcomic
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Readcomic : ParsedHttpSource() {
+
+    override val name = "ReadComic"
+
+    override val baseUrl = "https://readcomic.net"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override fun popularMangaSelector() = ".manga-box"
+
+    override fun latestUpdatesSelector() = ".home-list .hl-box .hlb-name"
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun popularMangaNextPageSelector() = ".general-nav a:last-child"
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/popular-comic/$page", headers)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic-updates/$page", headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET(
+            "$baseUrl/advanced-search".toHttpUrl().newBuilder().apply {
+                addQueryParameter("key", query)
+                if (!filters.isEmpty()) {
+                    for (filter in filters) {
+                        when (filter) {
+                            is StatusFilter -> {
+                                addQueryParameter("status", arrayOf("", "CMP", "ONG")[filter.state])
```

Comment:
add function/value getter in the StatusFilter class to return the correct value to add to url